File: DynamicProgramming/16climbingStairs/climbingStairs.py
# Climbing Stairs - Leetcode 70 (Dynammic programming)
class Solution:
    def climbStairs(self, n: int) -> int:
        one, two = 1, 1
        for i in range(n - 1):
            temp = one
            one = one + two
            two = temp
        return one
    
print(Solution().climbStairs(5))
print(Solution().climbStairs(4))    
print(Solution().climbStairs(3))    
print(Solution().climbStairs(2))

File: DynamicProgramming/16climbingStairs/climbingStairs.java

class climbingStairs {
    public static int ClimbStairs(int n) {
        int one = 1,
                two = 1,
                temp;
        for (int i = 0; i < n - 1; i++) {
            temp = one;
            one = one + two;
            
            two = temp;
        }
        return one;
    }

    public static void main(String[] args) {
        System.out.print(ClimbStairs(5));
    }
}